### PR TITLE
Add LCDproc

### PR DESCRIPTION
**Pull request by Treer**
- 308076b is a pfSense 2.3 ports package containing the original LCDproc files
- at 7fe021f the conversion of the UI to use the new OO Forms system is finished
- at 93f8cfa I've added [3 extra screens](https://forum.pfsense.org/index.php?topic=110177.msg648970#msg648970) to take advantage of 4+ row LCDs

One of the extra screens is displaying the output of `bandwidth_by_ip.php`, but that file is designed for ajax calls so requires an authenticated session, so this LCDproc package has its own copy of that file which it can run locally in the `usr/local/pkg` directory, which seems unfortunate but necessary, perhaps someone has a better idea?

I've compiled and used this as a [manually installed](https://github.com/treer/FreeBSD-ports#installation) ports package in pfSense 2.3.1-RELEASE-p5 and 2.3.3-DEVELOPMENT, with the new UI and screens, but don't know how to test it with pfSense's package manager, and have not made a package for pfSense before.

### CODE REVIEW

**Comment by Treer** [comment]
[hubba has also posted a fix for the CARP screen](https://forum.pfsense.org/index.php?topic=110177.msg652986#msg652986), I can add it to this pull request, but I'm linking to it now incase the pull request is merged first.

File: sysutils/pfSense-pkg-LCDproc/Makefile [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+RUN_DEPENDS=	${LOCALBASE}/sbin/LCDd:sysutils/lcdproc
+
+do-extract:
+	${MKDIR} ${WRKSRC}
+
+do-install:
+	${MKDIR} ${STAGEDIR}${PREFIX}/pkg
+	${MKDIR} ${STAGEDIR}/etc/inc/priv
+	${MKDIR} ${STAGEDIR}${PREFIX}/www/packages/lcdproc
```
Comment (rbgarga) [inline sysutils/pfSense-pkg-LCDproc/Makefile]: Whitespace before EOL, please remove

File: sysutils/pfSense-pkg-LCDproc/Makefile [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+PORTVERSION=	0.10.3
+CATEGORIES=	sysutils
+MASTER_SITES=	# empty
+DISTFILES=	# empty
+EXTRACT_ONLY=	# empty
+
+MAINTAINER=	coreteam@pfsense.org
+COMMENT=	LCDproc package for pfSense
+
+LICENSE=	ESF
```
Comment (rbgarga) [inline sysutils/pfSense-pkg-LCDproc/Makefile]: We moved from ESF to APACHE20 license

File: sysutils/pfSense-pkg-LCDproc/files/etc/inc/priv/lcdproc.priv.inc [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+	THIS SOFTWARE IS PROVIDED ``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES,
+	INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+	AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+	AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+	OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+	SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+	INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+	CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+	ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+	POSSIBILITY OF SUCH DAMAGE.
```
Comment (rbgarga) [inline sysutils/pfSense-pkg-LCDproc/files/etc/inc/priv/lcdproc.priv.inc]: Change license text to APACHE20

File: sysutils/pfSense-pkg-LCDproc/files/usr/local/pkg/lcdproc.xml [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+	AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+	AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+	OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+	SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+	INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+	CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+	ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+	POSSIBILITY OF SUCH DAMAGE.
+*/
+/* ====================================================================================== */
```
Comment (rbgarga) [inline sysutils/pfSense-pkg-LCDproc/files/usr/local/pkg/lcdproc.xml]: Please fix license text

File: sysutils/pfSense-pkg-LCDproc/files/usr/local/pkg/lcdproc.xml [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+	]]>
+	</copyright>
+	<title>Services: LCDproc</title>
+	<name>lcdproc</name>
+	<version>0.10.3</version>
```
Comment (rbgarga) [inline sysutils/pfSense-pkg-LCDproc/files/usr/local/pkg/lcdproc.xml]: Change 0.10.3 to %%PKGVERSION%% knob

File: sysutils/pfSense-pkg-LCDproc/Makefile [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+	${INSTALL_DATA} ${FILESDIR}${PREFIX}/www/packages/lcdproc/lcdproc_screens.php \
+		${STAGEDIR}${PREFIX}/www/packages/lcdproc
+	${INSTALL_DATA} ${FILESDIR}${PREFIX}/www/packages/lcdproc/index.php \
+		${STAGEDIR}${PREFIX}/www/packages/lcdproc
+	${INSTALL_DATA} ${FILESDIR}/etc/inc/priv/lcdproc.priv.inc \
+		${STAGEDIR}/etc/inc/priv
+	${INSTALL_DATA} ${FILESDIR}${DATADIR}/info.xml \
+		${STAGEDIR}${DATADIR}
+	@${REINPLACE_CMD} -i '' -e "s|%%PKGVERSION%%|${PKGVERSION}|" \
+		${STAGEDIR}${DATADIR}/info.xml
```
Comment (rbgarga) [inline sysutils/pfSense-pkg-LCDproc/Makefile]: Add all XML files that use %%PKGVERSION%% knob to this REINPLACE_CMD statement

File: sysutils/pfSense-pkg-LCDproc/files/usr/local/www/packages/lcdproc/lcdproc_screens.php [hunk 7]
```diff
@@ -0,0 +1,336 @@
...
+	THIS SOFTWARE IS PROVIDED ``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES,
+	INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+	AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+	AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+	OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+	SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+	INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+	CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+	ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+	POSSIBILITY OF SUCH DAMAGE.
```
Comment (rbgarga) [inline sysutils/pfSense-pkg-LCDproc/files/usr/local/www/packages/lcdproc/lcdproc_screens.php]: Fix license text

File: sysutils/pfSense-pkg-LCDproc/pkg-plist [hunk 8]
```diff
@@ -0,0 +1,10 @@
+pkg/lcdproc.xml
+pkg/lcdproc.inc
+pkg/lcdproc_client.php
+pkg/lcdproc_bandwidth_by_ip.php
+/etc/inc/priv/lcdproc.priv.inc
+www/packages/lcdproc/lcdproc.php
+www/packages/lcdproc/lcdproc_screens.php
+www/packages/lcdproc/index.php
+%%DATADIR%%/info.xml
+@dir /etc/inc/priv
```
Comment (rbgarga) [inline sysutils/pfSense-pkg-LCDproc/pkg-plist]: It's missing @dir /etc/inc here

**Comment by Treer** [comment]
I think GitHub's "Review Changes" function collapsed 3 of the files, causing then to be overlooked. They also needed license text fixed:
 * sysutils\pfSense-pkg-LCDproc\files\usr\local\pkg\lcdproc_client.php
 * sysutils\pfSense-pkg-LCDproc\files\usr\local\pkg\lcdproc.inc
 * sysutils\pfSense-pkg-LCDproc\files\usr\local\www\packages\lcdproc\lcdproc.php

I've fixed the license text for those 3 in a seperate commit - ad76a83

**Comment by rbgarga** [comment]
Added to 2.4.0 and 2.3.3 snapshots. After more people test it and it's ready, we can add to 2.3.2

**Comment by Treer** [comment]
By copying the files from the old package into this new ports package, I've managed to lose their git histories.

Is it possible to get the git commit message of [this commit](https://github.com/pfsense/FreeBSD-ports/commit/308076bdf95c35cb5863b73c50492d72eeba09e7) amended to include some sort of link to [the git repo the files came from](https://github.com/pfsense/pfsense-packages/tree/70312d161a7675288a274c1304839c352323f9eb/config/lcdproc-dev)?

**Pull request merged**